src/features/berth/selectedBerthPage/SelectedBerthPageContainer.tsx

```diff
@@ -76,21 +91,38 @@ const SelectedBerthPageContainer = ({
       render={({ handleSubmit, invalid, values, form: { change } }) => (
         <SelectedBerthPage
           applicationType={applicationType}
-          reasonOptions={reasonOptions}
           boatInfo={boatInfo}
-          change={change}
           deselectBerth={deselectBerth}
+          enableSubmit={
+            selectedHarbors.size === 0 ||
+            invalid ||
+            (applicationType === ApplicationOptions.SwitchApplication && !values.berth)
+          }
```

Comment:
Having these values extracted to their own variables would make the component more legible.